public/components/event_analytics/explorer/log_patterns/log_patterns.tsx

```diff
@@ -49,21 +49,35 @@ const EventPatterns = ({
     requestParams: { tabId },
   });
 
+  // refresh patterns on opening page
+  useEffect(() => {
+    getPatterns(selectedIntervalUnit?.value?.replace(/^auto_/, '') || 'y');
+  }, []);
+
   const onPatternSelection = async (pattern: string) => {
     if (query[FILTERED_PATTERN] === pattern) {
       return;
     }
-    dispatch(
+    // await here allows react to render update properly and display it.
+    // it forces the query to be changed before running it, without await the visual wont update.
+    await dispatch(
       changeQuery({
         tabId,
         query: {
           [FILTERED_PATTERN]: pattern,
         },
       })
     );
-    // workaround to refresh callback and trigger fetch data
-    await setTempQuery(query[RAW_QUERY]);
```

Comment:
same comment here about the tempquery - can we remove it all together? If it is necessary can we consolidate it into the query in redux instead of having a separate useState variable?

Reply:
Removed tempQuery from log patterns, It is still used in explorer to handle query searches and changes.